pkg/gadgets/helpers.go

```diff
@@ -232,3 +233,57 @@ func FixBpfKtimeGetBootNs(programSpecs map[string]*ebpf.ProgramSpec) {
 		removeBpfKtimeGetBootNs(s)
 	}
 }
+
+// OptsWithMountnsMap adds an ebpf.Map to the ebpf.CollectionOptions and creates
+// the ebpf.CollectionOptions if it does not exist
+func OptsWithMountnsMap(opts *ebpf.CollectionOptions, mountnsMap *ebpf.Map) *ebpf.CollectionOptions {
+	if opts == nil {
+		opts = &ebpf.CollectionOptions{}
+	}
+	if opts.MapReplacements == nil {
+		opts.MapReplacements = map[string]*ebpf.Map{}
+	}
+	if mountnsMap != nil {
+		opts.MapReplacements["mount_ns_filter"] = mountnsMap
+	}
+
+	return opts
+}
+
+// LoadeBPFSpec collects operations that are duplicated across gadgets.
+// It replaces filter map and calls the necessary function to load
+// Maps and Programs into the kernel
+func LoadeBPFSpec(
+	mountnsMap *ebpf.Map,
+	spec *ebpf.CollectionSpec,
+	consts map[string]interface{},
+	objs interface{},
+) error {
+	FixBpfKtimeGetBootNs(spec.Programs)
+
+	mapReplacements := map[string]*ebpf.Map{}
+	filterByMntNs := false
+
+	if mountnsMap != nil {
+		filterByMntNs = true
+		mapReplacements["mount_ns_filter"] = mountnsMap
```

Comment:
nit: The `"mount_ns_filter` -> `MntNsFilterMapName` should be done in this commit. (Not in the second one). However, you can squash both commits if you want to avoid dealing with this issue.

Reply:
Right, thanks. I moved it to the first one 😄